src/main/java/de/plushnikov/intellij/plugin/processor/clazz/log/AbstractSimpleLogProcessor.java

```diff
@@ -0,0 +1,54 @@
+package de.plushnikov.intellij.plugin.processor.clazz.log;
+
+import java.lang.annotation.Annotation;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import com.intellij.psi.PsiClass;
+import de.plushnikov.intellij.plugin.lombokconfig.ConfigDiscovery;
+import org.jetbrains.annotations.NotNull;
+
+abstract class AbstractSimpleLogProcessor extends AbstractLogProcessor {
+  @NotNull
+  private final String loggerType;
+  @NotNull
+  private final String loggerInitializer;
+  @NotNull
+  private final Function<Boolean, List<LoggerInitializerParameter>> loggerInitializerParameters;
```

Comment:
Do we really need this as function defined by constructor? Maybe we can make it simpler just by implementing the corresponding method in the derived class?

Reply:
I found a better way. Function was a too general solution indeed.